### PR TITLE
After validating token, redirects to change pw

### PR DESCRIPTION
**Pull request by juhangsin**

### CODE REVIEW

File: src/reset-password/controller/redirect.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import Controller from '@curveball/controller';
+
+class ChangePasswordController extends Controller {
+
+
+}
+
+export default new ChangePasswordController();
```
Comment (juhangsin) [inline src/reset-password/controller/redirect.ts]: Temporary file

File: src/app.ts [hunk 2]
```diff
@@ -30,4 +30,8 @@ import { load } from './server-settings';
 
   console.log('Listening on port', port);
 
+  if (!process.env.PUBLIC_URI) {
+    process.env.PUBLIC_URI = 'http://localhost:' + port + '/';
```
Comment (evert) [inline src/app.ts]: These new lines should go above `app.listen`, and I  think it should also have a console.log, like `PUBLIC_URI environment variable was not set, defaulting to x`

File: src/reset-password/service.ts [hunk 3]
```diff
@@ -43,24 +43,21 @@ export async function createToken(user: User): Promise<string> {
...
+        token,
         tokenTTL
     ]);
     return token;
 }
 
-export async function validateToken(user: User, token: string): Promise<boolean> {
-    const query = 'SELECT token FROM reset_password_token WHERE user_id = ? AND token = ? AND expires_at > UNIX_TIMESTAMP()';
-    const result = await database.query(query, [user.id]);
+export async function validateToken(token: string): Promise<User> {
```
Comment (evert) [inline src/reset-password/service.ts]: One of the requirements for this function is that tokens can only be used once. So when a token is 'valid', it should be deleted from the table.

File: src/reset-password/service.ts [hunk 4]
```diff
@@ -43,24 +43,21 @@ export async function createToken(user: User): Promise<string> {
...
+    const result = await database.query(query, [token]);
 
-    const hashes: string[] = result[0].map( (row: { token: string }) => row.token );
-
-    for (const hash of hashes) {
-        if (await bcrypt.compare(token, hash)) {
-            return true;
-        }
+    if (result[0].length !== 1) {
+      throw new BadRequest ('Invaild Token');
```
Comment (evert) [inline src/reset-password/service.ts]: Needs a better message

File: src/app.ts [hunk 5]
```diff
@@ -26,12 +26,14 @@ import { load } from './server-settings';
 
   const port = process.env.PORT ? parseInt(process.env.PORT, 10) :  8531;
 
+  if (!process.env.PUBLIC_URI) {
+    process.env.PUBLIC_URI = 'http://localhost:' + port + '/';
+    console.log('PUBLIC_URI environment variable was not set, defaulting to *localhost*')
```
[the remark this answers is not in the document]
Comment (juhangsin) [reply src/app.ts]: I'll fix this for the next PR!

**Pull request merged**